src/containers/Firmware/index.js

```diff
@@ -54,11 +54,19 @@ const Firmware = ({
 
   const createUpdateAssignment = ({ firmwareVersionRecordId, modelId }) => {
     const { createdTimestamp, lastModifiedTimestamp } = traskAssignmentValues;
+    let prevFirmwareVersionRecordId = firmwareVersionRecordId;
+    trackAssignmentData.forEach((element, index) => {
+      if (element.modelId === modelId) {
+        prevFirmwareVersionRecordId = trackAssignmentData[index].firmwareVersionRecordId;
+      }
+    });
```

Comment:
this is not how you should get it. You already know/have the data for the Assignment being updated